developer/js/libexec/create-override-script-regexp.ts

```diff
@@ -0,0 +1,134 @@
+for (let {codePoint, generalCategory} of unicodeData()) {
+  if (!block.contains(codePoint)) {
+    block = nextBlock();
+  }
+  console.assert(block.contains(codePoint));
+
+  if (SPACELESS_SCRIPT_BLOCKS.has(block.name) && isLetterOrMark(generalCategory)) {
+    elligibleCharacters.push(codePoint)
+  }
+}
+
+let ranges = groupCodePointsIntoRanges(elligibleCharacters)
+
+let characterClasses = ranges.map(([lower, upper]) => {
+  if (lower === upper) {
+    return unicodeEscape(lower);
+  } else if (lower == upper - 1) {
+    return unicodeEscape(lower) + unicodeEscape(upper);
+  } else {
+    return `${unicodeEscape(lower)}-${unicodeEscape(upper)}`;
+  }
+}).join("");
+
+console.log(`/**
+ * AUTOMATICALLY GENERATED FILE. DO NOT MODIFY
+ * See: libexec/create-override-script-regexp.ts for details!
+ */
+export const HAS_SOUTHEAST_ASIAN_LETTER = /[${characterClasses}]/;`);
+
+
+////////////////////////////////// Helpers ///////////////////////////////////
+
+function* unicodeData() {
+  let unicodeDataFile = readFileSync(path.join(UCD_DIR, "UnicodeData.txt"), "UTF-8");
+  for (let line of unicodeDataFile.split("\n")) {
+    if (line.trim() == "") {
+      continue;
+    }
+
+    let parts = line.split(";")
+    yield {
+      codePoint: parseInt(parts[0], 16),
+      generalCategory: parts[2],
+    }
+  }
+}
+
+function* blocks() {
+  let blocksFile = readFileSync(path.join(UCD_DIR, "Blocks.txt"), "UTF-8");
+  for (let line of blocksFile.split("\n")) {
+    if (line.trim() === "") {
+      continue;
+    }
+    if (line.startsWith("#")) {
+      continue;
+    }
+
+    let [range, name] = line.split("; ");
+    let [lower, upper] = range.split("..").map(s => parseInt(s, 16));
+
+    yield {
+      name, lower, upper,
+      contains(codePoint: number) {
+        return lower <= codePoint && codePoint <= upper;
+      }
+    }
+  }
+}
+
+function nextBlock() {
+  let {value} = blockIter.next();
+  if (!value) {
+    throw new Error("ran out of blocks");
+  }
+  return value;
+}
+
+function isLetterOrMark(category: string): boolean {
+  return category.startsWith("L") || category.startsWith("M");
+}
+
+function groupCodePointsIntoRanges(characters: number[]): [number, number][] {
+  let ranges = [];
+
+  let previousCharacter = characters[0];
+  let candidates = characters.slice(1);
+  let currentRange: [number, number] = [previousCharacter, previousCharacter];
+  for (let codePoint of candidates) {
+    if (codePoint === previousCharacter + 1) {
+      currentRange[1] = codePoint;
+    } else {
+      ranges.push(currentRange);
+      currentRange = [codePoint, codePoint];
+    }
+
+    previousCharacter = codePoint;
+  }
+
+  return ranges;
+}
+
+function unicodeEscape(codePoint: number) {
+  if (codePoint > 0xFFFF) {
+    throw new Error("non-BMP code points not supported");
+  }
+
+  let hex = codePoint.toString(16).toUpperCase();
+  let leadingZeros = "0".repeat(4 - hex.length);
+  return `\\u${leadingZeros}${hex}`;
```

Comment:
What, no left-pad? 🤣 (please, no, don't add a dependency for that!) 

```suggestion
  let hex = codePoint.toString(16).toUpperCase().padStart(4,'0');
  return `\\u${hex}`;
```

Reply:
I didn't know `.padStart()` was well supported, but apparently it's been in Node since version 8+!

Reply:
🤯